plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseJdbcConnectorTest.java

```diff
@@ -1146,4 +1161,118 @@ protected TestView createSleepingView(Duration minimalSleepDuration)
     {
         throw new UnsupportedOperationException();
     }
+
+    @Test
+    public void testMetadataDeleteWithBigintEquality()
+    {
+        if (!hasBehavior(SUPPORTS_DELETE)) {
+            assertQueryFails("DELETE FROM region WHERE regionkey = 1", "This connector does not support deletes");
+            return;
+        }
+        String tableName = "test_delete_" + randomTableSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " AS SELECT * FROM region", 5);
+        assertUpdate("DELETE FROM " + tableName + " WHERE regionkey = 1", 1);
+        assertQuery("SELECT count(*) FROM " + tableName, "VALUES 4");
+        assertUpdate("DROP TABLE " + tableName);
+    }
+
+    @Test
+    public void testMetadataDeleteWithVarcharEquality()
+    {
+        if (!hasBehavior(SUPPORTS_DELETE) || !hasBehavior(SUPPORTS_PREDICATE_PUSHDOWN_WITH_VARCHAR_EQUALITY)) {
+            assertQueryFails(
+                    "DELETE FROM nation WHERE name = 'ALBANIA'",
+                    "(.*[Dd]elet(e|ing).*(not |un)supported.*)" +
+                            "|(This connector does not support deletes)" +
+                            "|(Unsupported delete)");
```

Comment:
the expected message should be exact, since it's about a well defined case where predicate is not subsumed.

```java
if (!hasBehavior(SUPPORTS_DELETE)) {
 // handled in other test
 // OR assertQueryFails("DELETE FROM nation", "This connector does not support deletes");
 return;
}
if (!hasBehavior(SUPPORTS_PREDICATE_PUSHDOWN_WITH_VARCHAR_EQUALITY)) {
  ... this assertion, but with exact expected message ...
}
```